macros.py

```diff
@@ -239,11 +240,16 @@ def last_date(self):
         dates, fields = self._dates()
         return dates[-1]
 
-    def exploitable_on(self, date):
-        # A somewhat simplistic calculation at the moment
-        if self.first_date() <= date and self.last_date() > date:
-            return True
-        return False
+    def exploitable_on(self, date, version=None):
+        """Returns a boolean giving whether this vulnerability was exploitable on a given date, using a particular version of Android. If no version is given, it is only recorded as exploitable if no Android versions have yet received a patch for it"""
```

Comment:
Use some newlines here so that the line is not so long.

Reply:
Changed